phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewUtilIT.java

```diff
@@ -0,0 +1,315 @@
+        Properties tenantProps = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        tenantProps.setProperty(PhoenixRuntime.TENANT_ID_ATTRIB, tenantId);
+        TableName catalogOrChildTableName = ViewUtil.getSystemTableForChildLinks(0, config);
+        String schema = generateUniqueName();
+        byte[] schemaInBytes = schema.getBytes(StandardCharsets.UTF_8);
+        byte[] tenantIdInBytes = tenantId.getBytes(StandardCharsets.UTF_8);
+        byte[] emptyTenantIdInBytes = new byte[0];
+        String multiTenantTableName = schema + "." + generateUniqueName();
+        String globalViewName = schema + "." + generateUniqueName();
+        String tenantViewOnMultiTenantTable1 = schema + "." + generateUniqueName();
+        String viewName2 = generateUniqueName();
+        String tenantViewOnMultiTenantTable2 = schema + "." + viewName2;
+        String tenantViewIndex = viewName2 + "_INDEX";
+        String tenantViewOnGlobalView = schema + "." + generateUniqueName();
+
+        String multiTenantTableDDL = "CREATE TABLE " + multiTenantTableName + "(TENANT_ID CHAR(10) NOT NULL, " +
+                "ID CHAR(10) NOT NULL, NUM BIGINT CONSTRAINT PK PRIMARY KEY (TENANT_ID, ID))" +
+                " MULTI_TENANT=true";
+        String globalViewDDL = "CREATE VIEW " + globalViewName + "(PK1 BIGINT, PK2 BIGINT) AS SELECT * FROM "
+                + multiTenantTableName + " WHERE NUM > -1";
+        String viewDDL = "CREATE VIEW %s AS SELECT * FROM %s";
+        String viewIndexDDL = "CREATE INDEX " + tenantViewIndex + " ON " + tenantViewOnMultiTenantTable2 +
+                "(NUM DESC) INCLUDE (ID)";
+
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement().execute(multiTenantTableDDL);
+            conn.createStatement().execute(globalViewDDL);
+
+            try (Connection tenantConn = DriverManager.getConnection(getUrl(), tenantProps)) {
+                tenantConn.createStatement().execute(
+                        String.format(viewDDL, tenantViewOnGlobalView, globalViewName));
+                tenantConn.createStatement().execute(
+                        String.format(viewDDL, tenantViewOnMultiTenantTable1, multiTenantTableName));
+                tenantConn.createStatement().execute(
+                        String.format(viewDDL, tenantViewOnMultiTenantTable2, multiTenantTableName));
+                tenantConn.createStatement().execute(viewIndexDDL);
+            }
+
+            try (PhoenixConnection phoenixConnection = DriverManager.getConnection(getUrl(),
+                    props).unwrap(PhoenixConnection.class)) {
+                Table catalogOrChildTable = phoenixConnection.getQueryServices().getTable(
+                        SchemaUtil.getPhysicalName(catalogOrChildTableName.toBytes(),
+                                phoenixConnection.getQueryServices().getProps()).getName());
+
+                assertEquals(true, ViewUtil.hasChildViews(catalogOrChildTable,
```

Comment:
You can directly use assertTrue or assertFalse instead of assertEquals(true, expression) in all four places